### ISSUE TITLE
fake R fails grow_resource_db_jgf

### ISSUE THREAD

**Issue by dongahn** [flux-framework/flux-sched#828]
The following bug was discovered while testing https://github.com/flux-framework/flux-core/issues/3625#issuecomment-830183266
```
flux R encode -r '0-1' -c '0-64' | flux ion-R encode > fake.R.json
flux start
./R-feeder
flux module remove sched-fluxion-resource
flux module reload resource noverify
flux module load sched-fluxion-resource load-allowlist=node,core,gpu policy=first
2021-05-01T02:05:48.963973Z sched-fluxion-resource.err[0]: grow_resource_db_jgf: db.load: error inserting an edge to outedge metadata map: cluster0 -> quartz15; : Cannot allocate memory
2021-05-01T02:05:48.964257Z sched-fluxion-resource.err[0]: update_resource_db: grow_resource_db: Cannot allocate memory
2021-05-01T02:05:48.964297Z sched-fluxion-resource.err[0]: update_resource: update_resource_db: Cannot allocate memory
2021-05-01T02:05:48.964562Z sched-fluxion-resource.err[0]: populate_resource_db_acquire: update_resource: Cannot allocate memory
2021-05-01T02:05:48.964611Z sched-fluxion-resource.err[0]: populate_resource_db: loading resources using resource.acquire
```

**Comment by dongahn** [comment]
It turned out that the identical hostname in the fake R is a problem. 

As part of [graph editing for elasticity](https://github.com/flux-framework/flux-sched/blob/master/resource/readers/resource_reader_jgf.cpp#L478), we only add a new vertex when its path does not exist in the graph. So, with two identical hostname, the second vertex is evaluated as "existence" and that causes the growth failure. 

Secondly, we have been using `ENOMEM` for map insert failures. But it seems it is an existent key is a more common problem that leads to a failure. I will change that to `EEXIST`.

**Comment by dongahn** [comment]
Using `-H` option for `flux R encode` should work around the issue. 

```
ahn1@docker-desktop:/usr/src$ flux R encode -r '0-1' -c '0-64' -H host[1-2]
{"version": 1, "execution": {"R_lite": [{"rank": "0-1", "children": {"core": "0-64"}}], "starttime": 0.0, "expiration": 0.0, "nodelist": ["host[1-2]"]}}
```

**Comment by dongahn** [comment]
@milroy: could you comment whether this is worthwhile to fix?

**Comment by dongahn** [comment]
During manual testing for https://github.com/flux-framework/flux-sched,  I spent way too much time to debug a problem which ends up being caused by this deficiency. I will see if I can fix this as a separate PR. 

Two possibilities:
1. Change code to live without assuming a path be always unique
2. Make a path be unique by augmenting the hostname component of the path with uniq_id.

**Comment by dongahn** [comment]
This has been addressed by PR #924
